cpp/src/io/parquet/bloom_filter_reader.cu

```diff
@@ -0,0 +1,688 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "compact_protocol_reader.hpp"
+#include "io/parquet/parquet.hpp"
+#include "reader_impl_helpers.hpp"
+
+#include <cudf/ast/detail/expression_transformer.hpp>
+#include <cudf/ast/detail/operators.hpp>
+#include <cudf/ast/expressions.hpp>
+#include <cudf/detail/cuco_helpers.hpp>
+#include <cudf/detail/transform.hpp>
+#include <cudf/hashing/detail/xxhash_64.cuh>
+#include <cudf/logger.hpp>
+#include <cudf/utilities/span.hpp>
+#include <cudf/utilities/traits.hpp>
+#include <cudf/utilities/type_checks.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_buffer.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <cuco/bloom_filter_policies.cuh>
+#include <cuco/bloom_filter_ref.cuh>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/tabulate.h>
+
+#include <future>
+#include <numeric>
+#include <optional>
+
+namespace cudf::io::parquet::detail {
+namespace {
+
+/**
+ * @brief Converts bloom filter membership results (for each column chunk) to a device column.
+ *
+ */
+struct bloom_filter_caster {
+  cudf::device_span<cudf::device_span<cuda::std::byte> const> bloom_filter_spans;
+  host_span<Type const> parquet_types;
+  size_t total_row_groups;
+  size_t num_equality_columns;
+
+  enum class is_int96_timestamp : bool { YES, NO };
```

Comment:
Should this use an existing `enum class`? Maybe it doesn't matter since this is entirely internal to this file.

https://github.com/rapidsai/cudf/blob/dc2a75cba40d38f4a6ba66e652764e96fa6b593d/cpp/src/io/statistics/statistics_type_identification.cuh#L42